internal/characters/eula/burst.go

```diff
@@ -117,44 +118,42 @@ func (c *char) triggerBurst() {
 	c.burstCounter = 0
 }
 
-func (c *char) burstStacks() {
-	c.Core.Events.Subscribe(event.OnEnemyDamage, func(args ...interface{}) bool {
-		atk := args[1].(*combat.AttackEvent)
-		dmg := args[2].(float64)
-		if c.Core.Status.Duration(burstKey) == 0 {
-			return false
+// When Eula's own Normal Attack, Elemental Skill, and Elemental Burst deal DMG to opponents,
+// they will charge the Lightfall Sword, which can gain an energy stack once every 0.1s.
+func (c *char) makeBurstStackCB() combat.AttackCBFunc {
+	if c.Core.Status.Duration(burstKey) == 0 {
+		return nil
+	}
```

Comment:
Should this be pushed into the callback? In the current state not 1:1 to old behavior iff you burst while skill/attack is in-flight -- (old behavior check happens when attack lands, new behavior when attack gets queued)